tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolLookupTest.java

```diff
@@ -317,6 +317,31 @@ public void testSymbolLookupWithAnnotationOnFunction() {
         assert symbolStringList.containsAll(expectedNameList);
     }
 
+    @Test
+    public void testSymbolLookupInMatchClause() {
+        Project project = BCompileUtil.loadProject("test-src/symbol_lookup_in_match.bal");
+        Package currentPackage = project.currentPackage();
+        ModuleId defaultModuleId = currentPackage.getDefaultModule().moduleId();
+        PackageCompilation packageCompilation = currentPackage.getCompilation();
+        SemanticModel model = packageCompilation.getSemanticModel(defaultModuleId);
+        Document srcFile = getDocumentForSingleSource(project);
+
+        BLangPackage pkg = packageCompilation.defaultModuleBLangPackage();
+        ModuleID moduleID = new BallerinaModuleID(pkg.packageID);
+
+        Map<String, Symbol> symbolsInMappingMatch = getSymbolsInFile(model, srcFile, 21, 26, moduleID);
+        assertList(symbolsInMappingMatch, Arrays.asList("a", "func", "v", "x1", "x2"));
+
+        Map<String, Symbol> symbolsInTupleMatch = getSymbolsInFile(model, srcFile, 25, 26, moduleID);
+        assertList(symbolsInTupleMatch, Arrays.asList("a", "func", "v", "x3", "x4"));
+
+        Map<String, Symbol> symbolsInErrorMatch = getSymbolsInFile(model, srcFile, 29, 26, moduleID);
+        assertList(symbolsInErrorMatch, Arrays.asList("a", "func", "v", "x5", "x6"));
+
+        Map<String, Symbol> symbolsInVarMatch = getSymbolsInFile(model, srcFile, 33, 26, moduleID);
+        assertList(symbolsInVarMatch, Arrays.asList("a", "func", "v", "x7"));
```

Comment:
Shall we add some cases where the position is within the match pattern etc.? Also, I think these can be fed in through a data provider

Reply:
Ack

Reply:
@pubudu91 Any suggestions to add test cases for scenarios within the pattern?

Reply:
Nothing in particular. What I meant was, all the current positions in the test seem to be positions within the block statement of the match clause. My suggestion was to add some cases where the positions are within the match pattern of the clause as well.